dbt/include/athena/macros/materializations/seeds/helpers.sql

```diff
@@ -14,23 +14,109 @@
   {%- set s3_data_naming = config.get('s3_data_naming', target.s3_data_naming) -%}
   {%- set external_location = config.get('external_location', default=none) -%}
 
-  {% set sql %}
-    create external table {{ this.render_hive() }} (
+  {%- set s3_location = adapter.upload_seed_to_s3(
+    s3_data_dir,
+    s3_data_naming,
+    external_location,
+    model.schema,
+    model.name,
+    agate_table,
+  ) -%}
+
+  -- create tmp relation
+  {%- set tmp_relation = api.Relation.create(
+    identifier=model.name + "_tmp",
+    schema=model.schema,
+    database=model.database,
+    type='table'
+  ) -%}
+
+  -- create target relation
+  {%- set relation = api.Relation.create(
+    identifier=model.name,
+    schema=model.schema,
+    database=model.database,
+    type='table'
+  ) -%}
+
+  -- drop tmp relation if exists
+  {{ drop_relation(tmp_relation) }}
+
+  {% set sql_tmp_table %}
+    create external table {{ tmp_relation.render_hive() }} (
         {%- for col_name in agate_table.column_names -%}
-            {%- set inferred_type = adapter.convert_type(agate_table, loop.index0) -%}
-            {%- set type = column_override.get(col_name, inferred_type) -%}
             {%- set column_name = (col_name | string) -%}
-            {{ adapter.quote_seed_column(column_name, quote_seed_column) }} {{ type }} {%- if not loop.last -%}, {% endif -%}
+            {{ adapter.quote_seed_column(column_name, quote_seed_column) }} string {%- if not loop.last -%}, {% endif -%}
```

Comment:
Why the tmp table should have only string columns and not the real types ?

Reply:
because is hard to inherit the type just by reading the csv header.
Inferring the type implies reading part of the dataset using something like pandas/polars, and then setting up the type properly

Reply:
Oh yes indeed.

Reply:
@Jrmyy we could propagate some of the real types to the temp table, but `OpenSerdeCSV` is not great with `date` or `timestamp` types, it seemed safer to cast them as string and was inline with @nicor88 comment:
> create an tmp external table that point to the csv - all columns can be treated as string